src/System.Private.CoreLib/shared/System/IO/FileStream.Windows.cs

```diff
@@ -61,7 +61,14 @@ private void Init(FileMode mode, FileShare share)
             int fileType = Interop.Kernel32.GetFileType(_fileHandle);
             if (fileType != Interop.Kernel32.FileTypes.FILE_TYPE_DISK)
             {
+                int errorCode = fileType == Interop.Kernel32.FileTypes.FILE_TYPE_UNKNOWN ? Marshal.GetLastWin32Error() : Interop.Errors.ERROR_SUCCESS;
+
                 _fileHandle.Dispose();
+
+                if (errorCode != Interop.Errors.ERROR_SUCCESS)
+                {
+                    throw Win32Marshal.GetExceptionForWin32Error(errorCode);
```

Comment:
This changes the exception type thrown.  We're ok with that?

Reply:
GetFileType is going to fail on either:
- Invalid handle passed to it. It would be likely caused by handle lifetime bug in the program that can lead to arbitrary crashes, hangs or data corruptions.
- Transient resource availability issue (memory, network, etc.)

For these corner cases, I think it is fine to throw exception that matches the other file I/O error handling.

Reply:
Ok